core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -129,6 +146,31 @@ func (u *upkeepStateStore) Start(pctx context.Context) error {
 	return nil
 }
 
+func (u *upkeepStateStore) flush(ctx context.Context) {
+	u.sem <- struct{}{}
```

Comment:
nit. don't fully understand the reason for this, is it to limit at most 10 flushes at once? (10 is channel size)

Reply:
Yeah I didn't want to have an unbounded number of go routines running (since I'm not sure how frequently we'll need to flush to the DB, or how long those flushes can take) so I just ballparked at most 10 at a time. Happy to remove this if it's overkill!

Reply:
i think this will automatically be limited since flush is done in a loop? Don't have strong opinions ok to keep as is

Reply:
Bit the loop triggers the batch insert asynchronously?